cloudify_cli/bootstrap/bootstrap.py

```diff
@@ -322,3 +330,45 @@ def _copy_agent_key(agent_local_key_path, agent_remote_key_path,
     with fabric.settings(**fabric_env):
         fabric.put(agent_local_key_path, agent_remote_key_path, use_sudo=True)
     return agent_remote_key_path
+
+
+def _upload_resources(manager_node, fabric_env, management_ip, rest_port,
+                      protocol):
+
+    upload_resources = \
+        manager_node.properties['cloudify'].get('upload_resources')
+
+    # upload the plugin using CLI upload command
+    rest_client = \
+        utils.get_rest_client(management_ip, rest_port, protocol) \
+        if 'plugin_resources' in upload_resources else None
+
+    for plugin_archive_path in upload_resources.get('plugin_resources', ()):
+        utils.upload_plugin(file(plugin_archive_path), management_ip,
+                            rest_client, plugins.validate)
+
+    # uploading dsl resources
+    remote_plugins_folder = '/opt/manager/resources/spec/'
+
+    for dsl_resources in upload_resources.get('dsl_resources', ()):
+
+        plugin_local_yaml_path = dsl_resources.get('resource_local_path')
+
+        if plugin_local_yaml_path:
+            plugin_remote_yaml_path = \
+                dsl_resources.get('resource_remote_path', '')
+
+            # copy plugin's yaml file to the manager's fileserver
+            remote_paths = plugin_remote_yaml_path.split('/')
+            if not remote_paths[-1].endswith('.yaml'):
```

Comment:
do we really want the heuristic? let the user upload whatever he wants and don't make these decisions for him